packages/drivers/driver-base/src/driverUtils.ts

```diff
@@ -128,16 +129,38 @@ export function validateMessages(
 			messages.length = 0;
 		}
 		if (last + 1 !== from + length) {
+			// If not strict, then return the first consecutive sub-block.
+			if (strict) {
+				messages.length = 0;
+			} else {
+				let validOpsCount = 1;
+				while (
+					validOpsCount < messages.length &&
+					messages[validOpsCount].sequenceNumber ===
+						messages[validOpsCount - 1].sequenceNumber + 1
+				) {
+					validOpsCount += 1;
+				}
+				messages.length = Math.min(messages.length, validOpsCount);
+			}
+			let firstValidOpSeqNumber: number | undefined;
+			let lastValidOpSeqNumber: number | undefined;
+			if (messages.length > 0) {
+				firstValidOpSeqNumber = messages[0].sequenceNumber;
```

Comment:
I'm not sure logging firstValidOpSeqNumber is useful - it's always same as `from`.
I'd instead have too lengths - original one, and one after fixup